src/openforms/submissions/utils.py

```diff
@@ -71,8 +79,46 @@ def remove_submission_uploads_from_session(
         remove_upload_from_session(attachment.temporary_file, session)
 
 
+def get_confirmation_email_components(submission: Submission):
+    if (
+        submission.form.confirmation_email_option
+        == ConfirmationEmailOptions.form_specific_email
+    ):
+        email_template = submission.form.confirmation_email_template
+        subject = email_template.subject
+        html_content = render_confirmation_email_content(
+            submission, email_template.content
+        )
+        text_content = render_confirmation_email_content(
+            submission, email_template.content, {"rendering_text": True}
+        )
+        return subject, html_content, text_content
+    else:
+        config = GlobalConfiguration.get_solo()
+        subject = Template(config.confirmation_email_subject).render(
+            Context({"form_name": submission.form.name})
+        )
+        html_content = render_confirmation_email_content(
+            submission, config.confirmation_email_content
+        )
+        text_content = render_confirmation_email_content(
+            submission, config.confirmation_email_content, {"rendering_text": True}
+        )
+        return subject, html_content, text_content
+
+
 def send_confirmation_email(submission: Submission):
-    email_template = submission.form.confirmation_email_template
+    logevent.confirmation_email_start(submission)
+
+    if submission.form.confirmation_email_option == ConfirmationEmailOptions.no_email:
```

Comment:
it's weird that this check is smeared out across two functions - my initial feedback was to raise an exception if e-mail sending needs to be skipped for whatever reason, catch that exception and then abort the actual sending + log the skip.

That way you can keep things contained/scoped and easy to follow.